pkg/controller/cluster/main_controller.go

```diff
@@ -429,8 +444,8 @@ func (c *Controller) syncHandler(key string) error {
 	// number does not equal the current desired replicas on the StatefulSet, we
 	// should update the StatefulSet resource.
 	if mi.GetReplicas() != *ss.Spec.Replicas {
-		glog.V(4).Infof("MinIOInstance %s replicas: %d, StatefulSet replicas: %d", name, mi.GetReplicas(), *ss.Spec.Replicas)
-		ss = statefulsets.NewForCluster(mi, svc.Name)
+		klog.V(4).Infof("MinIOInstance %s replicas: %d, StatefulSet replicas: %d", name, mi.GetReplicas(), *ss.Spec.Replicas)
+		ss = statefulsets.NewForCluster(mi, hlSvc.Name)
 		err = c.kubeClientSet.AppsV1().StatefulSets(mi.Namespace).Delete(ctx, ss.Name, metav1.DeleteOptions{})
 		if err != nil {
```

Comment:
Should we rollback previous steps (svc creation, hsvc creation, etc)

Reply:
We attempt this steps until they succeed, that's why we check for the resource before we attempt.